src/components/App/App.jsx

```diff
@@ -0,0 +1,400 @@
+      // Add saved article to state if it was successfully added on the backend
+      setSavedArticles((prevArticles) => {
+        const isAlreadySaved = prevArticles.some(
+          (savedArticle) => savedArticle.articleId === likedArticle.articleId
+        );
+        // Avoid duplicate entries in the frontend state
+        return isAlreadySaved ? prevArticles : [...prevArticles, likedArticle];
+      });
+    })
+    .catch((err) => {
+      if (err.code === 11000) {
+        console.log("Article already exists in the database.");
+      } else {
+        console.error("Error in handleCardLike:", err);
+      }
+    });
+};
+
+  // // Handle saving (liking) articles
+  // const handleCardLike = (article) => {
+  //   const token = getToken();
+  //   if (!token) return;
+  //   // console.log(article);
+  //   const articleId = generateArticleId(article);
+  //   article.articleId = articleId;
+  //   // console.log(article.articleId);
+
+  //   // Attach the keywords (from searchQuery) to the article
+  //   article.keywords = searchQuery.split(" "); // Split search query into individual keywords
+
+  //   // console.log("Article with attached keywords:", article.keywords.slice(0,2)); // Log the article with keywords for confirmation
+
+  //   auth
+  //     .likeArticle(article, token) // Pass the articleId and article object to the API
+  //     .then((likedArticle) => {
+  //       setSavedArticles([...savedArticles, likedArticle]); // Add saved article to state
+  //     })
+  //     .catch(console.error);
+  // };
+
+  const handleCardDelete = (article) => {
+    const token = getToken();
+    if (!token) return;
+
+    // Find the matching article in savedArticles by comparing URLs
+    const savedArticle = savedArticles.find(
+      (saved) => saved.url === article.url
+    );
+
+    if (savedArticle) {
+      const articleId = savedArticle._id; // Use MongoDB _id from savedArticles
+
+      // Call the backend to delete the article by its _id
+      auth
+        .deleteArticle(articleId, token) // API call to delete the article by _id
+        .then(() => {
+          console.log("Article deleted:", articleId);
+          // Remove the article from savedArticles state
+          setSavedArticles((prevArticles) =>
+            prevArticles.filter((a) => a._id !== articleId)
+          );
+        })
+        .catch((error) => {
+          console.error("Error deleting article:", error);
+        });
+    } else {
+      console.log("No matching saved article found for deletion.");
+    }
+  };
+
+  const handleSignin = () => {
+    setActiveModal("sign-in");
+  };
+
+  const handleSignup = () => {
+    setActiveModal("sign-up");
+  };
+
+  const closeActiveModal = () => {
+    setActiveModal("");
+  };
+
+  const handleOpenModal = () => {
+    setIsModalOpen(true);
+  };
+
+  useEffect(() => {
+    const handleKeyDown = (e) => {
+      if (e.key === "Escape" && activeModal) {
+        closeActiveModal();
+      }
+    };
```

Comment:
Please, stop the effect if there is no active modal to correctly remove the listener


```js
   useEffect(() => {

    if (!activeModal) return; // stop the effect not to add the listener if there is no active modal

```